speaker/bgp_controller.go

```diff
@@ -335,13 +335,27 @@ func (c *bgpController) updateAds() error {
 		if peer.session == nil {
 			continue
 		}
-		if err := peer.session.Set(allAds...); err != nil {
+		ads := adsPerPeer(peer.cfg.Name, allAds)
+		if err := peer.session.Set(ads...); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+func adsPerPeer(peerName string, ads []*bgp.Advertisement) []*bgp.Advertisement {
+	res := []*bgp.Advertisement{}
+	for _, a := range ads {
+		if a.MatchesPeer(peerName) {
+			res = append(res, a)
+		}
+	}
+	if len(res) == 0 {
+		return nil
+	}
+	return res
```

Comment:
do we actually care if the slice is nil / empty later? if not I think just returning res is easier to understand

Reply:
This keeps the behaviour consistent with what we have today. If there are no ads, then we feed with nil. I can change if you think it doesn't matter, I kept on the conservative side (plus, I didn't want to fix a ton of tests :P )

Reply:
nvm then, just keep it this way 😄